test/inspector/test-bindings.js

```diff
@@ -97,10 +98,43 @@ function testSampleDebugSession() {
   assert.throws(() => session.post('Debugger.enable'), (e) => !!e);
 }
 
-testNoCrashWithExceptionInCallback();
-testSampleDebugSession();
-let breakpointHit = false;
-scopeCallback = () => (breakpointHit = true);
-debuggedFunction();
-assert.strictEqual(breakpointHit, false);
-testSampleDebugSession();
+async function testNoCrashConsoleLogBeforeThrow() {
+  const session = new inspector.Session();
+  session.connect();
+  let attempt = 1;
+  const promise = new Promise(function(resolve) {
+    let count = 0;
+    process.on('warning', () => {
```

Comment:
you can use `common.mustCall(3)` here to limit the number of warning events triggered.